handler/src/main/java/io/netty/handler/pcap/PCAPWriteHandler.java

```diff
@@ -0,0 +1,410 @@
+
+    /**
+     * {@code true} if we want to capture packets with zero bytes else {@code false}.
+     */
+    private final boolean captureZeroByte;
+
+    /**
+     * {@code true} if we want to write Pcap Global Header on initialization of
+     * {@link PCapWriter} else {@code false}.
+     */
+    private final boolean writePcapGlobalHeader;
+
+    /**
+     * TCP Sender Segment Number.
+     */
+    private int sendSegmentNumber = 1;
+
+    /**
+     * TCP Receiver Segment Number
+     */
+    private int receiveSegmentNumber = 1;
+
+    /**
+     * Source Address [TCP ONLY]
+     */
+    private InetSocketAddress srcAddr;
+
+    /**
+     * Destination Address [TCP ONLY]
+     */
+    private InetSocketAddress dstAddr;
+
+    /**
+     * Create new {@link PCAPWriteHandler} Instance.
+     * {@code captureZeroByte} is set to {@code false} and
+     * {@code writePcapGlobalHeader} is set to {@code true}.
+     *
+     * @param outputStream OutputStream where Pcap data will be written
+     */
+    public PCAPWriteHandler(OutputStream outputStream) {
+        this(outputStream, false, true);
+    }
+
+    /**
+     * Create new {@link PCAPWriteHandler} Instance
+     *
+     * @param outputStream          OutputStream where Pcap data will be written
+     * @param captureZeroByte       Set to {@code true} to enable capturing packets with empty (0 bytes) payload.
+     *                              Otherwise, if set to {@code false}, empty packets will be filtered out.
+     * @param writePcapGlobalHeader Set to {@code true} to write Pcap Global Header on initialization.
+     *                              Otherwise, if set to {@code false}, Pcap Global Header will not be written
+     *                              on initialization. This could when writing Pcap data on a existing file where
+     *                              Pcap Global Header is already present.
+     */
+    public PCAPWriteHandler(OutputStream outputStream, boolean captureZeroByte, boolean writePcapGlobalHeader) {
+        this.outputStream = outputStream;
+        this.captureZeroByte = captureZeroByte;
+        this.writePcapGlobalHeader = writePcapGlobalHeader;
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+
+        /*
+         * If `writePcapGlobalHeader` is `true`, we'll write Pcap Global Header.
+         */
+        if (writePcapGlobalHeader) {
+
+            ByteBuf byteBuf = ctx.alloc().buffer();
+            try {
+                this.pCapWriter = new PCapWriter(this.outputStream, byteBuf);
+            } catch (IOException ex) {
+                ctx.fireExceptionCaught(ex);
+            } finally {
+                byteBuf.release();
+            }
+
+        } else {
+            this.pCapWriter = new PCapWriter(this.outputStream);
+        }
+
+        if (ctx.channel() instanceof SocketChannel) {
+            if (ctx.channel() instanceof ServerChannel) {
```

Comment:
I think this is not correct... Shouldn't this be `ctx.channel().parent()` ?

Reply:
Alright. Still learning the internals of Netty so didn't know. Fixed :D